pycargoebuild/__main__.py

```diff
@@ -71,13 +72,30 @@ def main(prog_name: str, *argv: str) -> int:
     load_license_mapping(args.license_mapping)
     args.license_mapping.close()
 
+    def iterate_parents(directory: Path) -> typing.Generator[Path, None, None]:
+        root = directory.absolute().root
+        yield directory
+        while not directory.samefile(root):
+            directory /= ".."
+            yield directory
```

Comment:
I believe we should also test `/Cargo.lock` presence, that is why this generator does not look same as you most probably expected (with respect to your last suggestion). This tool can be used in some container, for example, where `Cargo.lock` can be located in root. Just to advocate my thoughts.

Reply:
WFM ;-).